test/lib/mock-blpapi.ts

```diff
@@ -0,0 +1,279 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+import assert = require('assert');
+import events = require('events');
+import util = require('util');
+import Promise = require('bluebird');
+import _ = require('lodash');
+import blpapi = require('blpapi');
+
+// Re-export blpapi public interfaces
+/* tslint:disable:interface-name */
+export interface SessionOpts extends blpapi.SessionOpts {}
+export interface Subscription extends blpapi.Subscription {}
+export interface Identity extends blpapi.Identity {}
+/* tslint:enable:interface-name */
+export interface ISession extends blpapi.ISession {}
+
+export interface IInstruction {
+    start?: boolean;
+    openService?: boolean;
+    stop?: boolean;
+    request?: string[];
+
+    [index: string]: any;
+}
+
+// ipc channel through websocket
+export var ipc: SocketIO.Socket;
+export var instructions: IInstruction = {};
+
+// Mapping of request types to response names to listen for.
+// The strings are taken from section A of the BLPAPI Developer's Guide, and are organized by
+// service.
+var REQUEST_TO_RESPONSE_MAP: { [index: string]: string; } = {
+    // //blp/refdata
+    'HistoricalDataRequest': 'HistoricalDataResponse',
+    'IntraDayTickRequest':   'IntraDayTickResponse',
+    'IntraDayBarRequest':    'IntraDayBarResponse',
+    'ReferenceDataRequest':  'ReferenceDataResponse',
+    'PortfolioDataRequest':  'PortfolioDataResponse',
+    'BeqsRequest':           'BeqsResponse',
+
+    // //blp/apiflds
+    'FieldInfoRequest':              'fieldResponse',
+    'FieldSearchRequest':            'fieldResponse',
+    'CategorizedFieldSearchRequest': 'categorizedFieldResponse',
+
+    // //blp/instruments
+    'instrumentListRequest': 'InstrumentListResponse',
+    'curveListRequest':      'CurveListResponse',
+    'govtListRequest':       'GovtListResponse',
+
+    // //blp/tasvc
+    'studyRequest':          'studyResponse'
+};
+
+// Mapping of service URIs to the event names to listen to when subscribed to these services.
+var SERVICE_TO_SUBSCRIPTION_EVENTS_MAP: { [uri: string]: string[]; } = {
+    '//blp/mktdata':  ['MarketDataEvents'],
+    '//blp/mktvwap':  ['MarketDataEvents'],
+    '//blp/mktbar':   ['MarketBarStart', 'MarketBarUpdate', 'MarketBarEnd'],
+    '//blp/pagedata': ['PageUpdate']
+};
+
+function getServiceForSecurity(security: string): string
```

Comment:
Also shared (and maybe other code as well, but just another comment since the previous said "maps" and there's more than just maps).